jbpm-designer-backend/src/main/java/org/jbpm/designer/server/indexing/bpmn2/DesignerProcessDataEventListener.java

```diff
@@ -325,4 +334,10 @@ private void resolveUnqualifiedClasses() {
         }
         return unqualifiedClasses;
     }
+
+    public <T> Predicate<T> distinctByKey(Function<? super T, Object> keyExtractor) {
+        Map<Object, Boolean> map = new ConcurrentHashMap<>();
+        return t -> map.putIfAbsent(keyExtractor.apply(t),
```

Comment:
Wouldn't this be better with set instead of map? And then maybe use that tracking approach in the for loop here https://github.com/kiegroup/jbpm-designer/pull/656/files#diff-c6676cfce924296267db48d25744b7b8R257 so there won't be any need for stream processing which can save some time...